### ISSUE TITLE
DiscoverAsync strange behaviour between Debug sessions

### ISSUE THREAD

**Issue by szilardx** [lextudio/sharpsnmplib#51]
I am experiencing a strange behaviour in Visual Studio 2015.
If I use the DiscoveryAsync method, as presented in the samples (+ .ConfigureAwait(false) to avoid deadlocks), I get many  devices, the program obeys the interval settings - so it works well. 
Then if I press stop, and then start again the Debugging session, from console output results I can see that only a few devices are found, and the method does not obey the interval setting.

I tested this behaviour with the non async version of the method, and this is not present there. 

Could it be the case that I am doing something wrong, or can I use the Async version in a production code reliably?

Thanks in advance! 👍

**Comment by lextm** [comment]
You might check if there are changes in packets first,

http://docs.sharpsnmp.com/en/latest/tutorials/troubleshooting.html

**Comment by szilardx** [comment]
I checked with Wireshark. I can clearly see the get-request from my computer to the broadcast address, and then 19 get-response from the local network devices - in both case.
In the first c# debug session in Visual Studio 2015 I can see the AgentFound event being triggered aroud 19 times.
But in the second debug session (no code change) the AgentFound event only happens 3-7 times, and also the breakpoint is being hit earlier than 16 seconds. (The 16 seconds I set is waited on the first run.)
If I restart the Viusal Studio, then the DiscoverAsync works as expected - 19 devices found, 16 sec wait etc.
